src/coprocessor/dag/expr/builtin_cast.rs

```diff
@@ -74,15 +74,15 @@ impl FnCall {
         if val.is_none() {
             return Ok(None);
         }
-        let val = val.unwrap();
-
-        if !val.is_empty() && val[0] == b"-"[0] {
+        let val = try!(String::from_utf8(val.unwrap()));
```

Comment:
You don't need string here. Use `is_ascii_whitespace` to check bytes should be enough.